wcfsetup/install/files/lib/system/user/notification/UserNotificationHandler.class.php

```diff
@@ -303,7 +304,28 @@ public function fireEvent(
                     }
                 }
             }
-
+            $conditionBuilder = new PreparedStatementConditionBuilder();
+            $conditionBuilder->add('userID IN (?)', [\array_keys($recipients)]);
+            $sql = "SELECT userID, workerID
+                    FROM   wcf1_service_worker " . $conditionBuilder;
+            $statement = WCF::getDB()->prepare($sql);
+            $statement->execute($conditionBuilder->getParameters());
+            $map = $statement->fetchMap('userID', 'workerID', false);
+            $jobs = [];
+
+            foreach ($map as $userID => $workerIDs) {
+                $notification = $notifications[$userID]['object'] ?? null;
+                if ($notification === null) {
+                    continue;
+                }
+                \assert($notification instanceof UserNotification);
+                foreach ($workerIDs as $workerID) {
+                    $jobs[] = new ServiceWorkerDeliveryBackgroundJob($workerID, $notification->notificationID);
+                }
+            }
+            if ($jobs !== []) {
+                BackgroundQueueHandler::getInstance()->enqueueIn($jobs);
+            }
```

Comment:
it's legal to pass an empty array.
```suggestion
            BackgroundQueueHandler::getInstance()->enqueueIn($jobs);
```